src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -4770,17 +4769,15 @@ VOID MethodTableBuilder::TestMethodImpl(
     {
         BuildMethodTableThrowException(IDS_CLASSLOAD_MI_NONVIRTUAL_DECL);
     }
-    if (!IsMdVirtual(dwImplAttrs))
+    if (!IsMdVirtual(dwImplAttrs) && !IsMdStatic(dwImplAttrs))
     {
+        // TODO: better error message? When non-virtual, it must be static.
         BuildMethodTableThrowException(IDS_CLASSLOAD_MI_MUSTBEVIRTUAL);
     }
     // Virtual methods cannot be static
-    if (IsMdStatic(dwDeclAttrs))
```

Comment:
We should not permit static virtual methods on classes/structs. By our ecma design, those should continue to throw.